src/main.py

```diff
@@ -30,14 +31,66 @@ def handle_invalid_usage(error):
 def sitemap():
     return generate_sitemap(app)
 
-@app.route('/user', methods=['GET'])
-def handle_hello():
+@app.route('/users/<int:id>', methods=['GET'])
+def get_user(id):
+    if User.get_user(id) is None:
+        raise APIException('User not found', status_code=404)
+    else:
+        return User.get_user(id)
+
+@app.route('/login', methods=['GET'])
+def get_user_by_email():
+    body=request.get_json()
+
+    if User.get_user_by_email(body["email"],body["password"]) is "email":
+        raise APIException("email doesn't exist", status_code=404)
+    elif User.get_user_by_email(body["email"],body["password"]) is "password":
+        raise APIException("password for that email doesn't exist", status_code=404)
+    else:
+        return User.get_user_by_email(body["email"],body["password"])
+
+@app.route('/users/<int:id>/socials', methods=['GET'])
+def get_all_socials(id):
+    if Social.get_all_socials(id) is None:
+        raise APIException('Social media accounts not found', status_code=404)
+    else:
+
+        return jsonify(Social.get_all_socials(id))
+
+@app.route('/users/<int:id_user>/socials/<int:id_social>', methods=['GET'])
+def get_social(id_user,id_social):
+    if Social.get_social(id_user,id_social) is None:
+        raise APIException('Social media account not found', status_code=404) 
+    else:
+        return jsonify(Social.get_social(id_user,id_social))
```

Comment:
```suggestion
    social = Social.get_social(id_user,id_social)
    if  not social:
        raise APIException('Social media account not found', status_code=404) 
    return jsonify(social)
```
Cuidado con las llamadas repetidas a la base de datos a la larga degradan mucho el rendimiento de la aplicación.